controllers/operator/apigateway_controller.go

```diff
@@ -133,6 +135,15 @@ func (r *APIGatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	if oryOathkeeperStatus := r.oathkeeperReconciler.ReconcileAndVerifyReadiness(ctx, r.Client, &apiGatewayCR); !oryOathkeeperStatus.IsReady() {
+		// Ressetkk: ErrMinimumReplicasAvailable should not be exported here
+		// because it comes from internal package...
+		if errors.Is(oryOathkeeperStatus.NestedError(), oathkeeper.ErrMinimumReplicasAvailable{}) ||
+			apierrors.IsNotFound(oryOathkeeperStatus.NestedError()) {
+			// oathkeeper takes some time to reconcile
+			// if we hit expected error silently requeue reconciliation without throwing stacktrace
+			r.log.Info("ory is in not found or deployment is not ready, requeue silently", "error", oryOathkeeperStatus.NestedError())
```

Comment:
I was thinking about it. I think that NotFound should be explicitly taken care of. You don't really want any specific Error to be reported when ORY is in expected state so the user (and monitoring) doesn't get bombarded with errors that are not expected. With that the CR will be stuck in processing state. If any other error that is not handled gets reported, it's instantly flagged as Error in a CR.

Reply:
Otherwise I'd have to write complete error handling of ORY oathkeeper so I can react properly to an error oath keeper reconciliation gives me

Reply:
I hope that APIServer would return proper error if something is up with the deployment.

Reply:
If minimum replica won't get available, we will be stuck in this state and currently our alerting is based on Error as far as I know. I don't think that we will get any other return from the API Server in this case.
I would be cautious with this change, since it can easily break the whole monitoring.

Reply:
But then the error will be on every reconciliation loop until it gets processed unless you go through all pods, collect their state and form a proper error. I don't know if I'd like to go that way for now.